### PR TITLE
Improve Shortcut

### PR DESCRIPTION
**Pull request by NathanReb**
I added a Shortcut.t type so that I could simplify a bit `Gen_builder`. There's probably room for more improvement/sharing as there's still a bit of fields parsing logic in `Gen_builder` to find attr and loc fields but that can be done later!

This will also make it easier to write `Gen_viewer`!

### CODE REVIEW

**Comment by ghost** [comment]
I'm not too sure about the names `from` and `to_`. What about `outter` and `inner`?

**Comment by NathanReb** [comment]
Yeah I'm not too fond of `to_` and `from`. `outter` and `inner` sound good. I was also wondering about `record` and `variant`, it's a bit less generic but it leaves no room to interpretation and given we already mention fields I don't think it can hurt.

What do you think?

**Comment by NathanReb** [comment]
I went for record and variant but I'm happy to change it to `outter` and `inner` if you prefer!

**Comment by ghost** [comment]
One last suggestion: `inner_variant` and `outter_record`, just to express the relationship between the two. What do you think?

**Comment by NathanReb** [comment]
Sounds perfect! I'l change that!

**Comment by NathanReb** [comment]
Done!

File: ast/cinaps/shortcut.mli [hunk 1]
```diff
@@ -0,0 +1,32 @@
+(** Type for shortcuts from description types to their parent record types:
+    - [record] is the parent type name
+    - [variant] is the description type name
```
Comment (ghost) [inline ast/cinaps/shortcut.mli]: The should say `outter_record` and `outter_variant`

**Review by ghost — APPROVED** [review]
Looks good!

**Pull request merged**